### PR TITLE
RD-2750 Remove jQuery code deps

### PR DESCRIPTION
**Pull request by stepek**
Removed last jQuery code usage.

### CODE REVIEW

File: app/app.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ import Routes from './containers/Routes';
 import translation from './translations/en.json';
 import LoaderUtils from './utils/LoaderUtils';
 
-window.$ = $;
+window.$ = $; // Needed by topology widget.
```
Comment (Gelio) [inline app/app.tsx:53]: Minor:

If that's the case, maybe we could move this line to the topology widget itself? This way it will be added closer to where it's used

File: test/jest/setup.ts [hunk 2]
```diff
@@ -33,10 +32,6 @@ Object.keys(document.defaultView).forEach(property => {
     }
 });
 
-global.$ = $;
```
Comment (Gelio) [inline test/jest/setup.ts:36]: Yay :smile:

File: app/utils/SplashLoadingScreen.ts [hunk 3]
```diff
@@ -1,24 +1,30 @@
...
-
-const SPLASH_DIV = $('div.splashPage');
+const SPLASH_DIV: HTMLElement | null = document.querySelector('div.splashPage');
 const ACTIVE_CLASS = 'active';
-const FADE_OUT_SPEED = 'slow';
+const FADE_OUT_SPEED = 300;
 
 export default class SplashLoadingScreen {
+    static fadeOut(element: HTMLElement) {
+        element.style.setProperty('transition', `all ${FADE_OUT_SPEED}ms linear`);
```
Comment (Gelio) [inline app/utils/SplashLoadingScreen.ts:7]: If I'm not mistaken, we can make the code simpler by adding the `transition` in CSS (https://github.com/cloudify-cosmo/cloudify-stage/blob/stepek/RD-2750/backend/templates/style.ejs). Then, we would not have to remove it manually.

Also, how about being more specific and only transitioning the `opacity`, instead of `all`? :slightly_smiling_face:

File: app/utils/SplashLoadingScreen.ts [hunk 4]
```diff
@@ -1,24 +1,30 @@
...
+            element.style.removeProperty('opacity');
+            element.style.setProperty('display', 'none');
+        }, FADE_OUT_SPEED);
+    }
+
     static turnOn() {
-        if (!SPLASH_DIV.hasClass(ACTIVE_CLASS)) {
-            SPLASH_DIV.show();
-            SPLASH_DIV.addClass(ACTIVE_CLASS);
+        if (SPLASH_DIV && !SPLASH_DIV.classList.contains(ACTIVE_CLASS)) {
```
Comment (Gelio) [inline app/utils/SplashLoadingScreen.ts:17]: You could add 

```ts
if (!SPLASH_DIV) {
    throw new Error('Splash page element not found');
}
```

before the class declaration which would make checking if `SPLASH_DIV` is defined unnecessary :slightly_smiling_face:

File: app/utils/SplashLoadingScreen.ts [hunk 5]
```diff
@@ -1,24 +1,30 @@
-// @ts-nocheck File not migrated fully to TS
-/**
- * Created by jakubniezgoda on 06/13/2017.
- */
-
-const SPLASH_DIV = $('div.splashPage');
+const SPLASH_DIV: HTMLElement | null = document.querySelector('div.splashPage');
```
Comment (Gelio) [inline app/utils/SplashLoadingScreen.ts:1]: Minor:

The type annotation seems redundant to me, as it's the same as the default one for `querySelector`

```suggestion
const SPLASH_DIV = document.querySelector('div.splashPage');
```

File: app/utils/SplashLoadingScreen.ts [hunk 6]
```diff
@@ -1,24 +1,30 @@
...
 const ACTIVE_CLASS = 'active';
-const FADE_OUT_SPEED = 'slow';
+const FADE_OUT_SPEED = 300;
 
 export default class SplashLoadingScreen {
+    static fadeOut(element: HTMLElement) {
```
Comment (Gelio) [inline app/utils/SplashLoadingScreen.ts:6]: Can we make it a `private` method? Also, this always will refer to `SPLASH_DIV`, so we could ditch the parameter now :slightly_smiling_face: 

```suggestion
    private static fadeOut() {
```

File: app/utils/SplashLoadingScreen.ts [hunk 7]
```diff
@@ -1,24 +1,30 @@
...
+        element.style.setProperty('transition', `all ${FADE_OUT_SPEED}ms linear`);
+        element.style.setProperty('opacity', '0');
```
Comment (Gelio) [inline app/utils/SplashLoadingScreen.ts:8]: How about adding the `opacity` manipulation to the `.active` class in CSS? Then, we would not need to take care of it manually, and all this function would do it add `display: none` after a timeout :smile:

**Review by Gelio — COMMENTED** [review]
Nice TS migrations :+1: I appreciate those

File: app/components/layout/ScrollToTop.ts [hunk 8]
```diff
@@ -1,19 +1,15 @@
...
-    componentDidUpdate(prevProps) {
+class ScrollToTop extends Component<RouteComponentProps> {
+    componentDidUpdate(prevProps: RouteComponentProps) {
         const { location } = this.props;
+
         if (location !== prevProps.location) {
-            $('div.main').scrollTop(0);
+            window.scrollTo(0, 0);
         }
     }
```
Comment (qooban) [inline app/components/layout/ScrollToTop.ts:14]: We don't have a test for that scrolling functionality IIRC, so I assume you verified it and works fine, right?

File: app/utils/SplashLoadingScreen.ts [hunk 9]
```diff
@@ -1,24 +1,30 @@
-// @ts-nocheck File not migrated fully to TS
-/**
- * Created by jakubniezgoda on 06/13/2017.
- */
-
-const SPLASH_DIV = $('div.splashPage');
+const SPLASH_DIV: HTMLElement | null = document.querySelector('div.splashPage');
 const ACTIVE_CLASS = 'active';
-const FADE_OUT_SPEED = 'slow';
+const FADE_OUT_SPEED = 300;
```
Comment (qooban) [inline app/utils/SplashLoadingScreen.ts:3]: As none of these consts are exported, probably all of them can be lowercased to be aligned with https://github.com/airbnb/javascript#naming--uppercase .

(we started following AirBnB long after this file was created)

File: app/app.tsx (hunk 1 above)
Comment (qooban) [reply app/app.tsx:53]: Good idea. Let's move it as close to the usage as possible to avoid potential confusion in the future.
As leaving it here suggests that the whole application is using jQuery.

**Review by qooban — COMMENTED** [review]
Added some comments.

File: app/utils/SplashLoadingScreen.ts (hunk 7 above)
Comment (stepek) [reply app/utils/SplashLoadingScreen.ts:8]: No, because we have only animation of fade out.

File: app/components/layout/ScrollToTop.ts (hunk 8 above)
Comment (stepek) [reply app/components/layout/ScrollToTop.ts:14]: it's works but i'll change it to use only `div.main` container.

File: app/utils/SplashLoadingScreen.ts [hunk 10]
```diff
@@ -1,24 +1,39 @@
...
         }
     }
 
     static turnOff() {
-        if (SPLASH_DIV.hasClass(ACTIVE_CLASS)) {
-            SPLASH_DIV.fadeOut(FADE_OUT_SPEED);
-            SPLASH_DIV.removeClass(ACTIVE_CLASS);
+        if (!splashDiv) {
+            throw new Error('Splash page element not found');
+        }
```
Comment (qooban) [inline app/utils/SplashLoadingScreen.ts:32]: I see it is common for all methods, maybe we can extract it to avoid duplication?

Optional:

Actually it is very unlikely it will happen as `div.splashPage` is part of the `index.html` from the very early stage of the application running time, so maybe that check should be done only before `SplashLoadingScreen` class definition?

File: widgets/topology/src/Topology.tsx [hunk 11]
```diff
@@ -6,6 +6,8 @@ import { createExpandedTopology } from './DataProcessor';
 import ScrollerGlassHandler from './ScrollerGlassHandler';
 import TerraformDetailsModal from './TerraformDetailsModal';
 
+window.$ = $;
+
```
Comment (qooban) [inline widgets/topology/src/Topology.tsx:10]: This will work only if we have https://github.com/cloudify-cosmo/cloudify-blueprint-topology/pull/162 implemented and cloudify-blueprint-topology version updated in stage, right?

**Review by qooban — COMMENTED** [review]
Commented. 

I assume that this PR needs to wait for https://github.com/cloudify-cosmo/cloudify-blueprint-topology/pull/162 and possibly https://github.com/cloudify-cosmo/cloudify-stage/pull/1493 (depending on in which PR you'd like to update `package.json`).

File: widgets/topology/src/Topology.tsx [hunk 12]
```diff
@@ -6,6 +6,8 @@ import { createExpandedTopology } from './DataProcessor';
 import ScrollerGlassHandler from './ScrollerGlassHandler';
 import TerraformDetailsModal from './TerraformDetailsModal';
 
+window.$ = $;
```
Comment (Gelio) [inline widgets/topology/src/Topology.tsx:9]: Wait a second :smile: This looks like a magic import to me at the moment. Where do we get the `$` from? Should we not `import $ from 'jquery';` or something? If not, then maybe jQuery is already a global and this assignment is no necessary (which I would not be a fan of :smile:)?

Comment (Gelio) [reply widgets/topology/src/Topology.tsx:9]: Actually, since `cloudify-blueprint-topology` will have its own instance of jQuery thanks to https://github.com/cloudify-cosmo/cloudify-blueprint-topology/pull/162, maybe we can skip this line completely and remove jQuery altogether from stage?

`cloudify-blueprint-topology` could import jQuery itself and expose it globally if it wants to

File: app/utils/SplashLoadingScreen.ts (hunk 10 above)
Comment (Gelio) [reply app/utils/SplashLoadingScreen.ts:32]: I see that does not work :worried: 

image

I believe a helper method/function with such an `if` should do the job here :+1:

File: app/utils/SplashLoadingScreen.ts (hunk 7 above)
Comment (Gelio) [reply app/utils/SplashLoadingScreen.ts]: Ok, makes sense, thanks :+1:

**Review by Gelio — COMMENTED** [review]
The code looks mostly good to me. I want to only clarify the location and necessity of `window.$ = $` before approving :slightly_smiling_face:

File: app/utils/SplashLoadingScreen.ts (hunk 10 above)
Comment (stepek) [reply app/utils/SplashLoadingScreen.ts:32]: nop, the same errors about possibility of null 
image

Comment (Gelio) [reply app/utils/SplashLoadingScreen.ts:32]: You need to use an assertion function:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

**Comment by stepek** [comment]
`window.$ = $` can be removed after https://github.com/cloudify-cosmo/cloudify-stage/pull/1493

File: app/utils/SplashLoadingScreen.ts [hunk 13]
```diff
@@ -1,12 +1,16 @@
...
+
 const splashDiv = document.querySelector<HTMLElement>('div.splashPage');
 const activeClass = 'active';
 const fadeOutSpeed = 300;
 
 export default class SplashLoadingScreen {
     private static fadeOut() {
         if (!splashDiv) {
-            throw new Error('Splash page element not found');
+            log.error('Splash page element not found');
```
Comment (Gelio) [inline app/utils/SplashLoadingScreen.ts:10]: Why is this changed to an `log.error` instead of a `throw`? What's the reason we want to fail silently :smile: ? IMO if such an error happens in the app, we should not even continue as something is really broken and should be fixed by a developer

**Review by Gelio — COMMENTED** [review]
https://github.com/cloudify-cosmo/cloudify-stage/pull/1500#discussion_r673741492 is still unresolved :slightly_smiling_face:

File: app/utils/SplashLoadingScreen.ts (hunk 13 above)
Comment (stepek) [reply app/utils/SplashLoadingScreen.ts:10]: Failing tests unit tests in many places

Comment (Gelio) [reply app/utils/SplashLoadingScreen.ts:10]: Sooo... maybe we should get our unit tests setup to more closely resemble the actual UI, instead of changing the runtime logic to accommodate the tests? :slightly_smiling_face: I'm ok with creating a task to revert this change and add a `splashDiv` in the jsdom in unit tests :slightly_smiling_face:

**Review by qooban — COMMENTED** [review]
I updated the title to have JIRA ID reference. 
I have no more specific comments, but I'm assuming that there are some discussions not resolved, yet, so I'll not approve yet.

From my side: 
1. Please verify if all tests are passing.
2. I assume that the last PR for jQuery full removal will be https://github.com/cloudify-cosmo/cloudify-stage/pull/1493 right?

**Comment by stepek** [comment]
> 1. Please verify if all tests are passing.

System tests: 
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/979/



> 2. I assume that the last PR for jQuery full removal will be [RD-267  Remove jQuery-UI (#1473) #1493](https://github.com/cloudify-cosmo/cloudify-stage/pull/1493) right?

Yes, right

File: app/utils/SplashLoadingScreen.ts (hunk 13 above)
Comment (stepek) [reply app/utils/SplashLoadingScreen.ts:10]: Ok, ticket [RD-2841](https://cloudifysource.atlassian.net/browse/RD-2841).

File: widgets/topology/src/Topology.tsx (hunk 12 above)
Comment (stepek) [reply widgets/topology/src/Topology.tsx:9]: No, upgrade will be in https://github.com/cloudify-cosmo/cloudify-stage/pull/1493

Comment (Gelio) [reply widgets/topology/src/Topology.tsx:9]: Alright :+1:

**Comment by Gelio** [comment]
I am irritated by https://github.com/cloudify-cosmo/cloudify-stage/pull/1500#discussion_r673741492 not being addressed for 6 days despite other communication happening in this PR :frowning_face: Other than that comment, it looks good to me

File: app/utils/SplashLoadingScreen.ts (hunk 10 above)
Comment (stepek) [reply app/utils/SplashLoadingScreen.ts]: Fixed but added try catches to prevent failing tests

File: widgets/topology/src/Topology.tsx (hunk 11 above)
Comment (stepek) [reply widgets/topology/src/Topology.tsx:10]: no, Changes regarding cloudify-cosmo/cloudify-blueprint-topology added in separate RP

File: app/utils/SplashLoadingScreen.ts (hunk 10 above)
Comment (Gelio) [reply app/utils/SplashLoadingScreen.ts]: Thanks :+1:

File: widgets/topology/src/Topology.tsx (hunk 11 above)
Comment (Gelio) [reply widgets/topology/src/Topology.tsx:10]: What's the reason to update `cloudify-blueprint-topology` in a separate PR instead of here? Doing it here would allow us to remove this unnecessary change, if I'm not mistaken

**Review by Gelio — DISMISSED** [review]
Generally looks good to me :slightly_smiling_face:

File: widgets/topology/src/Topology.tsx (hunk 11 above)
Comment (stepek) [reply widgets/topology/src/Topology.tsx:10]: in next PR i'll remove jquery jquery ui and upgrade `cloudify-blueprint-topology`

**Review by qooban — APPROVED** [review]

File: widgets/topology/src/widget.tsx [hunk 14]
```diff
@@ -3,11 +3,20 @@
...
+import 'jquery-ui/ui/widgets/mouse';
+import 'jquery-ui/ui/widgets/draggable';
+import 'jquery-ui/ui/widgets/droppable';
+
 import { castArray } from 'lodash';
 import Topology from './Topology';
 import DataFetcher from './DataFetcher';
 import { createBaseTopology } from './DataProcessor';
 
+window.$ = $;
```
Comment (stepek) [inline widgets/topology/src/widget.tsx:18]: Mover from `Topology.tsx` to be initialized before importing `cloudify-blueprint-topology`

File: widgets/topology/src/widget.tsx [hunk 15]
```diff
@@ -3,11 +3,20 @@
  * Created by kinneretzin on 07/09/2016.
  */
 
+import $ from 'jquery';
+import 'jquery-ui/ui/core';
+import 'jquery-ui/ui/widget';
```
Comment (stepek) [inline widgets/topology/src/widget.tsx:8]: Moved closure to topology widget. will be removed in next pr with upgrade of `cloudify-blueprint-topology` and removement  jQuery

**Comment by stepek** [comment]
### system tests 
https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/995/

File: widgets/topology/src/widget.tsx (hunk 15 above)
Comment (qooban) [reply widgets/topology/src/widget.tsx:8]: I assume you are talking about this PR https://github.com/cloudify-cosmo/cloudify-stage/pull/1493 ?

🙂 Why not remove it at once within this PR?

**Review by qooban — APPROVED** [review]
I think it would be easier to remove all deps within one PR (especially that #1493 seems very small), but I'm also fine with having two as this is hopefully last phase of the removal, I suspect 🙂

File: widgets/topology/src/widget.tsx (hunk 15 above)
Comment (stepek) [reply widgets/topology/src/widget.tsx:8]: because I have one open :)
